app/src/main/java/org/mozilla/focus/fragment/BrowserFragment.java

```diff
@@ -1050,6 +1051,11 @@ private boolean isPopupWindowAllowed() {
         return ScreenNavigator.get(getContext()).isBrowserInForeground();
     }
 
+    private boolean isTabRestoredComplete() {
+        return (getActivity() instanceof MainActivity)
```

Comment:
NIT: throw an exception if not instanceof MainActivity so we won't have a false that does not really mean tab is restoring?

Reply:
Good idea! Refined.